src/app_errors/mod.rs

```diff
@@ -133,22 +133,22 @@ pub enum AppErrorKind {
 
     /// An error for when a bounce violation happens.
     #[fail(display = "Email account sent complaint")]
-    BounceComplaintError {
+    ComplaintError {
         address: EmailAddress,
         bounced_at: u64,
-        bounce: BounceRecord,
+        bounce: DeliveryProblem,
```

Comment:
I haven't changed these property names because it makes the handling code less uniform. I figure consumers will have an easier time if they're consistent with the bounce errors. I could easily be persuaded otherwise, though.

Reply:
(we should make our minds up about it soon though, because this is part of the published API)

Reply:
Could they be kept uniform by using a more generic term like "deliveryProblemType" and "deliveryProblemOccurredAt" or similar, in line with your renaming of the module?